File: macos-app/Lexume/Models/TargetLanguage.swift
import Foundation

struct TargetLanguage: Sendable, Equatable {
    var displayName: String
    var code: String
    var isRTL: Bool

    /// Persian stays first (the existing default). The rest cover most of
    /// what Google Translate/Gemini actually support — the original list of
    /// 10 was just an arbitrary carryover from the web app's scope, not a
    /// technical limit.
    static let all: [TargetLanguage] = [
        TargetLanguage(displayName: "Persian", code: "fa", isRTL: true),
        TargetLanguage(displayName: "Spanish", code: "es", isRTL: false),
        TargetLanguage(displayName: "French", code: "fr", isRTL: false),
        TargetLanguage(displayName: "German", code: "de", isRTL: false),
        TargetLanguage(displayName: "Chinese (Simplified)", code: "zh-CN", isRTL: false),
        TargetLanguage(displayName: "Chinese (Traditional)", code: "zh-TW", isRTL: false),
        TargetLanguage(displayName: "Japanese", code: "ja", isRTL: false),
        TargetLanguage(displayName: "Korean", code: "ko", isRTL: false),
        TargetLanguage(displayName: "Russian", code: "ru", isRTL: false),
        TargetLanguage(displayName: "Arabic", code: "ar", isRTL: true),
        TargetLanguage(displayName: "Turkish", code: "tr", isRTL: false),
        TargetLanguage(displayName: "Italian", code: "it", isRTL: false),
        TargetLanguage(displayName: "Portuguese", code: "pt", isRTL: false),
        TargetLanguage(displayName: "Dutch", code: "nl", isRTL: false),
        TargetLanguage(displayName: "Polish", code: "pl", isRTL: false),
        TargetLanguage(displayName: "Swedish", code: "sv", isRTL: false),
        TargetLanguage(displayName: "Norwegian", code: "no", isRTL: false),
        TargetLanguage(displayName: "Danish", code: "da", isRTL: false),
        TargetLanguage(displayName: "Finnish", code: "fi", isRTL: false),
        TargetLanguage(displayName: "Greek", code: "el", isRTL: false),
        TargetLanguage(displayName: "Hebrew", code: "he", isRTL: true),
        TargetLanguage(displayName: "Hindi", code: "hi", isRTL: false),
        TargetLanguage(displayName: "Urdu", code: "ur", isRTL: true),
        TargetLanguage(displayName: "Bengali", code: "bn", isRTL: false),
        TargetLanguage(displayName: "Vietnamese", code: "vi", isRTL: false),
        TargetLanguage(displayName: "Thai", code: "th", isRTL: false),
        TargetLanguage(displayName: "Indonesian", code: "id", isRTL: false),
        TargetLanguage(displayName: "Malay", code: "ms", isRTL: false),
        TargetLanguage(displayName: "Filipino", code: "tl", isRTL: false),
        TargetLanguage(displayName: "Ukrainian", code: "uk", isRTL: false),
        TargetLanguage(displayName: "Czech", code: "cs", isRTL: false),
        TargetLanguage(displayName: "Romanian", code: "ro", isRTL: false),
        TargetLanguage(displayName: "Hungarian", code: "hu", isRTL: false),
        TargetLanguage(displayName: "Swahili", code: "sw", isRTL: false),
        TargetLanguage(displayName: "Pashto", code: "ps", isRTL: true),
        TargetLanguage(displayName: "Kurdish (Sorani)", code: "ckb", isRTL: true),
    ]

    static func named(_ name: String) -> TargetLanguage {
        all.first(where: { $0.displayName == name }) ?? all[0]
    }
}

File: android-app/app/src/main/java/com/aboooooo57/lexume/data/model/TargetLanguage.kt
package com.aboooooo57.lexume.data.model

/** Kotlin analog of macOS's `Models/TargetLanguage.swift`. */
data class TargetLanguage(val displayName: String, val code: String, val isRtl: Boolean) {
    companion object {
        /**
         * Persian stays first (the existing default). The rest cover most of
         * what Google Translate/Gemini actually support - the original list
         * of 10 was just an arbitrary carryover from the web app's scope,
         * not a technical limit.
         */
        val all: List<TargetLanguage> = listOf(
            TargetLanguage("Persian", "fa", isRtl = true),
            TargetLanguage("Spanish", "es", isRtl = false),
            TargetLanguage("French", "fr", isRtl = false),
            TargetLanguage("German", "de", isRtl = false),
            TargetLanguage("Chinese (Simplified)", "zh-CN", isRtl = false),
            TargetLanguage("Chinese (Traditional)", "zh-TW", isRtl = false),
            TargetLanguage("Japanese", "ja", isRtl = false),
            TargetLanguage("Korean", "ko", isRtl = false),
            TargetLanguage("Russian", "ru", isRtl = false),
            TargetLanguage("Arabic", "ar", isRtl = true),
            TargetLanguage("Turkish", "tr", isRtl = false),
            TargetLanguage("Italian", "it", isRtl = false),
            TargetLanguage("Portuguese", "pt", isRtl = false),
            TargetLanguage("Dutch", "nl", isRtl = false),
            TargetLanguage("Polish", "pl", isRtl = false),
            TargetLanguage("Swedish", "sv", isRtl = false),
            TargetLanguage("Norwegian", "no", isRtl = false),
            TargetLanguage("Danish", "da", isRtl = false),
            TargetLanguage("Finnish", "fi", isRtl = false),
            TargetLanguage("Greek", "el", isRtl = false),
            TargetLanguage("Hebrew", "he", isRtl = true),
            TargetLanguage("Hindi", "hi", isRtl = false),
            TargetLanguage("Urdu", "ur", isRtl = true),
            TargetLanguage("Bengali", "bn", isRtl = false),
            TargetLanguage("Vietnamese", "vi", isRtl = false),
            TargetLanguage("Thai", "th", isRtl = false),
            TargetLanguage("Indonesian", "id", isRtl = false),
            TargetLanguage("Malay", "ms", isRtl = false),
            TargetLanguage("Filipino", "tl", isRtl = false),
            TargetLanguage("Ukrainian", "uk", isRtl = false),
            TargetLanguage("Czech", "cs", isRtl = false),
            TargetLanguage("Romanian", "ro", isRtl = false),
            TargetLanguage("Hungarian", "hu", isRtl = false),
            TargetLanguage("Swahili", "sw", isRtl = false),
            TargetLanguage("Pashto", "ps", isRtl = true),
            TargetLanguage("Kurdish (Sorani)", "ckb", isRtl = true)
        )

        fun named(name: String): TargetLanguage = all.firstOrNull { it.displayName == name } ?: all[0]
    }
}
